### ISSUE TITLE
4.2.2 release requires an update to the wazuh_api_yml template. manifest/manager.pp logic

### ISSUE THREAD

**Issue by jhowe-uw** [wazuh/wazuh-puppet#396]
Pull https://github.com/wazuh/wazuh/pull/10219 renamed /var/ossec/api/configuration/api.yaml => https::ssl_cipher to https::ssl_protocol.

This requires an update to the templates/wazuh_api_yml.erb file ( conditional based on wazuh version ? )

Environment:

Debian                                     10.10

wazuh-manager                     4.2.2-1
puppet-agent                         6.24.0-1buster

Expected:

puppet run to succeed

Actual:

puppet run failed

```
Error: Systemd start for wazuh-manager failed!
journalctl log for wazuh-manager:
-- Logs begin at Tue 2021-09-28 12:18:20 PDT, end at Tue 2021-09-28 12:23:35 PDT. --
Sep 28 12:23:21 vm systemd[1]: Starting Wazuh manager...
Sep 28 12:23:34 vm env[9556]: Configuration not valid: 2000 - Some parameters are not expected in the configuration file (WAZUH_PATH/api/configuration/api.yaml): Additional properties are not allowed ('ssl_cipher' was unexpected).
Sep 28 12:23:35 vm env[9556]: wazuh-apid: Configuration error. Exiting
Sep 28 12:23:35 vm systemd[1]: wazuh-manager.service: Control process exited, code=exited, status=1/FAILURE
Sep 28 12:23:35 vm systemd[1]: wazuh-manager.service: Failed with result 'exit-code'.
Sep 28 12:23:35 vm systemd[1]: Failed to start Wazuh manager.
```

Work-around

Disable puppet agent runs after initial deploy ( undesired ! ) , corrected api.yaml, resurrect wazuh-manager

Over-riding the api template by redefining wazuh_api_template via manifests/manager.pp doesn't work, since the template is hard-coded on line 663 in manager.pp

Redefining wazuh_api_ssl_cipher doesn't work either, as it still expects an ssl_protocol entry in api.yaml to work properly with a ssl_cipher definition without a ssl_protocol.

Tested ssl_cipher with  "" and "ECDHE+AESGCM:!ECDSA"

**Comment by jakefeasey** [comment]
Fixed and PR submitted: #397

**Comment by vcerenu** [comment]
Hi @jhowe-uw 

The error you sent us in version 4.2.2 was replicated
Thanks to the input of @jakefeasey we have already incorporated a fix for this error.
I appreciate the contribution of both and do not hesitate to create a new Issue or PR on any other topic
